### PR TITLE
Implement deadlines RFC.

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

**Review by hayley-jean — APPROVED** [review]
Just a small comment, but not a blocker for me

File: esdb/configuration.go [hunk 1]
```diff
@@ -70,6 +70,8 @@ type Configuration struct {
 
 	// The amount of time (in milliseconds) the sender of the keep alive ping waits for an acknowledgement.
 	KeepAliveTimeout time.Duration // Defaults to 10 seconds.
+
+	DefaultDeadline *time.Duration
```
[the remark this answers is not in the document]
Comment (YoEight) [reply esdb/configuration.go:76]: Done

**Review by hayley-jean — APPROVED** [review]

File: esdb/configuration.go [hunk 2]
```diff
@@ -70,6 +70,10 @@ type Configuration struct {
 
 	// The amount of time (in milliseconds) the sender of the keep alive ping waits for an acknowledgement.
 	KeepAliveTimeout time.Duration // Defaults to 10 seconds.
+
+	// The amount of time (in milliseconds) a non-streaming operation should take to complete before resulting in a
+	// DeadlineExceeded. Defaults to 10 seconds.
+	DefaultDeadline *time.Duration
```
Comment (pvanbuijtene) [inline esdb/configuration.go:76]: Is there a need for using a pointer here?
It creates a bit of an inconsistency with `KeepAliveInterval` & `KeepAliveTimeout`

Comment (YoEight) [reply esdb/configuration.go:76]: To detect if a default timeout was set.

**Review by pvanbuijtene — APPROVED** [review]

**Pull request merged**